src/Kiota.Builder/Refiners/GoRefiner.cs

```diff
@@ -130,8 +137,155 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
             );
             RemoveHandlerFromRequestBuilder(generatedCode);
             AddContextParameterToGeneratorMethods(generatedCode);
+            CorrectTypes(generatedCode);
         }, cancellationToken);
     }
+    private string MergeOverLappedStrings(string start, string end)
+    {
+        start = start.ToFirstCharacterUpperCase();
+        end = end.ToFirstCharacterUpperCase();
+        var endPattern = end[..(end.IndexOf("RequestBuilder", StringComparison.CurrentCultureIgnoreCase) + "RequestBuilder".Length)];
+
+        if (start.EndsWith(endPattern))
+            return $"{start[..start.IndexOf(endPattern)]}{end}";
+            
+        return $"{start}{end}";
+    }
+
+    private static void CorrectTypes(CodeElement currentElement)
+    {
+        if (currentElement is CodeMethod currentMethod && currentMethod.IsOfKind(CodeMethodKind.RequestBuilderBackwardCompatibility, CodeMethodKind.RequestBuilderWithParameters) && currentElement.Parent is CodeClass parentClass)
```

Comment:
```suggestion
        if (currentElement is CodeMethod currentMethod && currentMethod.IsOfKind(CodeMethodKind.RequestBuilderBackwardCompatibility, CodeMethodKind.RequestBuilderWithParameters) && currentElement.Parent is CodeClass)
```
Out of curiosity, do we need to check if the parent is a `CodeClass`? In my head a `CodeMethod` should always have a CodeClass as parent.